noir-projects/noir-contracts/contracts/claim_contract/src/main.nr

```diff
@@ -11,6 +11,7 @@ contract Claim {
     use dep::uint_note::uint_note::UintNote;
     use token::Token;
 
+    // TODO: This can be optimized by storing the addresses in Config struct in 1 PublicImmutable (less merkle proofs).
```

Comment:
This seems like quite a common optimization pattern.

Reply:
Yes, it'd be good for public immutable to explicitly mention on its docs that it is optimized for this (both source code and docsite).